app/src/ui/lib/filter-list.tsx

```diff
@@ -1,11 +1,15 @@
 import * as React from 'react'
 import * as classnames from 'classnames'
 
+import { Branch } from '../../models/branch'
```

Comment:
Let's keep this component generic. This component allows us to filter any list of items, not just branches. All of the state you need access to should be available from within the `BranchesContainer` component.